### PR TITLE
fix reading request body after request has been sent

### PR DESCRIPTION
**Pull request by vaidasm**
This fixes the way stream contents is read. See:
http://docs.guzzlephp.org/en/latest/streams.html#creating-streams

### CODE REVIEW

**Comment by odino** [comment]
hey @vaidasm thanks for the PR! Would you mind sharing a test case that was failing?

cheers!

**Comment by vaidasm** [comment]
Hi @odino, just added a simple testcase for this. `$request->getBody()->getContents()` is getting executed when request is being sent and when called the second time `getContents` returns empty string.

File: tests/Formatter/CurlFormatterTest.php [hunk 1]
```diff
@@ -107,4 +107,15 @@ public function testPUT()
         $this->assertContains("-d 'foo=bar&hello=world'", $curl);
         $this->assertContains("-X PUT", $curl);
     }
+
+    public function testProperBodyReading()
+    {
+        $request = new Request('PUT', 'example.local', [], Stream::factory('foo=bar&hello=world'));
+        $request->getBody()->getContents();
```
Comment (odino) [inline tests/Formatter/CurlFormatterTest.php]: hey @vaidasm what happens if we dont call this method?

**Comment by vaidasm** [comment]
`getContents` is reads stream from current position until the end. If we call it before sending request, position is 0, so this method returns all the body string. After guzzle send the request, position is not zero (its at the end of stream) - and so, `getContents` returns empty string. The proper way of getting the entire body string is to cast body to string (or call `__toString()`). 

You can test this behaviour: just run this test against current master.

**Comment by odino** [comment]
ah got it! then thanks man, we should merge it

**Pull request merged**